src/encoder.js

```diff
@@ -35,27 +35,43 @@ Encoder.prototype._e = function(jObj,e_schema){
 }
 
 Encoder.prototype.processValue= function(str,r){
-    if(!this.isAppChar(r[0]) && !this.isAppChar(str[str.length -1])){
+    if(!this.isAppChar(r[0]) && (!this.isAppChar(str[str.length -1]) 
+        || (this.isAppChar(str[str.length -1]) && str.length>1 
+            && str[str.length - 2] === '\\'))){
         str += chars.boundryChar;
     }
     return str + r;
 }
 
 /**
  * 
- * @param {*} a 
+ * @param {*} value 
  * @param {*} type 
  * @return {string} return either the parsed value or a special char representing the value
  */
-Encoder.prototype.getValue= function(a,type){
-    switch(a){
+Encoder.prototype.getValue= function(value,type){
+    switch(value){
         case undefined: return chars.missingPremitive;
         case null: return chars.nilPremitive;
         case "": return chars.emptyValue;
-        default: return this.dataHandlers[type].parse(a);
+        default: return sanitizeData(this.dataHandlers[type].parse(value));
     }
 }
 
+/**
+ * 
+ * @param {string} data 
+ * @return {string} return the sanitized string by adding backslash to the special chars
+ */
+function sanitizeData(data) {
+    if(typeof data === "string"){
+        appCharsArr.forEach(c => {
+            data = data.replace(new RegExp(c, 'g'), '\\'+c);
```

Comment:
I believe this loop can be avoided with range match.

Reply:
updated now